src/base/utils/sys.cpp

```diff
@@ -0,0 +1,356 @@
+/*
+ * Bittorrent Client using Qt and libtorrent.
+ * Copyright (C) 2018  Vladimir Golovnev <glassez@yandex.ru>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project's "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ */
+
+#include "sys.h"
+
+#ifdef Q_OS_WIN
+#include <shlobj.h>
+#include <shobjidl.h>
+#include <shobjidl_core.h>
+#include <winreg.h>
+
+#include <QDir>
+#include <QSettings>
+#include <QSysInfo>
+
+#define __WIDE(str) L ## str
+#define WIDE(str) __WIDE(str)
+
+#define APP_REGISTRY_NAME WIDE(APPLICATION_NAME)
+
+namespace
+{
+    enum REG_SEARCH_TYPE
+    {
+        USER,
+        SYSTEM_32BIT,
+        SYSTEM_64BIT
+    };
+
+    QStringList getRegSubkeys(HKEY handle)
+    {
+        QStringList keys;
+
+        DWORD cSubKeys = 0;
+        DWORD cMaxSubKeyLen = 0;
+        LONG res = ::RegQueryInfoKeyW(handle, NULL, NULL, NULL, &cSubKeys, &cMaxSubKeyLen, NULL, NULL, NULL, NULL, NULL, NULL);
+
+        if (res == ERROR_SUCCESS) {
+            cMaxSubKeyLen++; // For null character
+            LPWSTR lpName = new WCHAR[cMaxSubKeyLen];
```

Comment:
I'll ask you your own question: "Are you kidding?" I just moved most of this code from other (inappropriate) place. I'm not going to change it!

Reply:
Especially since I'm not going to do your inappropriate whims. All this code is written in WinAPI style and it uses its types and definitions. It worked fine for many years before, so why touch it?

Reply:
> I just moved most of this code from other (inappropriate) place

Then the copyright in this new file lies.

Reply:
Oh, sorry. I'll fix it. 
This PR is still quite unfinished...

Reply:
> All this code is written in WinAPI style...

As if it is something that has a right to live... :D

Reply:
>As if it is something that has a right to live...

It's none of our business...